tests/conftest.py

```diff
@@ -557,29 +564,45 @@ async def mqtt_mock(hass, mqtt_client_mock, mqtt_config):
     entry = MockConfigEntry(
         data=mqtt_config,
         domain=mqtt.DOMAIN,
-        title="Tasmota",
+        title="MQTT",
     )
-
     entry.add_to_hass(hass)
-    # Do not forward the entry setup to the components here
-    with patch("homeassistant.components.mqtt.PLATFORMS", []):
-        assert await hass.config_entries.async_setup(entry.entry_id)
+
+    real_mqtt = mqtt.MQTT
+    real_mqtt_instance = None
+    mock_mqtt_instance = None
+
+    async def _setup_mqtt_entry():
+        """Set up the MQTT config entry."""
+        if entry.state == ConfigEntryState.NOT_LOADED:
```

Comment:
I don't think we should have this check. If users of the fixture needs a different behavior they should use a custom fixture.

Reply:
Maybe two fixtures named `mqtt_mock_entry_with_yaml_config` and `mqtt_mock_entry_no_yaml_config`, that's the difference between tests needing one or the other version?